src/java/org/apache/cassandra/io/sstable/DefaultStorageHandler.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.sstable;
+
+import java.util.Collection;
+
+import org.apache.cassandra.db.Directories;
+import org.apache.cassandra.db.lifecycle.Tracker;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+import org.apache.cassandra.schema.TableMetadataRef;
+import org.apache.cassandra.service.StorageService;
+
+/**
+ * The default storage handler, used when sstables are stored on the local file system.
+ */
+public class DefaultStorageHandler extends StorageHandler
+{
+    public DefaultStorageHandler(TableMetadataRef metadata, Directories directories, Tracker dataTracker)
+    {
+        super(metadata, directories, dataTracker);
+    }
+
+    @Override
+    public boolean isReady()
+    {
+        return !StorageService.instance.isBootstrapMode();
```

Comment:
this method is meant to replay `isBootstrapMode` check in read path, see https://github.com/riptano/cndb/issues/2561

Reply:
ported #2561 here 0b3ba8c71dd134e8a57516109c349907d1d8e32b

Reply:
I see, though its return values can be surprising because the node initially is not bootstrapping, then it enters bootstrap, switches the flag, and switches it back when bootstrap if done, so the method would initially return `true`, then `false`, then `true` - is that ok? 

Wouldn't it be better to either use `StorageService.initialized` flag, or if it is set too late, add a new flag which would be set after bootstrap/no-bootstrap in `StorageService.joinTokenRing` is done?

Reply:
> I see, though its return values can be surprising because the node initially is not bootstrapping, then it enters bootstrap, switches the flag, and switches it back when bootstrap if done, so the method would initially return true, then false, then true - is that ok?

Agreed, It could happen if the method is used by bootstrap. But for current usage, it seems  safe.

> Wouldn't it be better to either use StorageService.initialized flag, or if it is set too late, add a new flag which would be set after bootstrap/no-bootstrap in StorageService.joinTokenRing is done?

make sense, we can open a issue in OSS.